pkg/apiserver/resthandler.go

```diff
@@ -684,15 +686,36 @@ func DeleteResource(r rest.GracefulDeleter, checkBody bool, scope RequestScope,
 // to use it.
 // TODO: add appropriate structured error responses
 func queryToObject(query url.Values, scope RequestScope, kind string) (runtime.Object, error) {
-	versioned, err := scope.Creater.New(scope.ServerAPIVersion, kind)
+	// TODO ListOptions is a mess, but there are pulls in progress to try to fix it to be unversioned.
+	// That would still end up special cased here, but at least it would be more sane.  We could check for
+	// unversioned things.
+	gvString := scope.APIVersion
+	internalGVString := scope.InternalVersion.String()
+
+	versioned, err := scope.Creater.New(gvString, kind)
 	if err != nil {
-		// programmer error
-		return nil, err
+		// if we have a failure for ListOptions in particular, then we should attempt to use the
+		// default ListOptions.  this is being addressed as we speak, so it should get better
+		if kind == "ListOptions" {
+			gvString = "v1"
```

Comment:
or remove ServerAPIVersion, hardcode "v1" (which preserves current behavior), and add to the list to clean up and allow decoding to unversioned